### DISCUSSION TITLE
WorkPool#unlimit why change all channels queue capacity size

### DISCUSSION THREAD

**Discussion by wangqiandeniangniang** [rabbitmq/rabbitmq-java-client#688] [Ideas]
When using java client version 5.11.0 , I feel confused，com.rabbitmq.client.impl.WorkPool#unlimit why change all channels queue capacity size ？why not change single channel ? 

 public synchronized void unlimit(K key) {
        unlimited.add(key);
        if (!unlimited.isEmpty()) {
            setCapacities(Integer.MAX_VALUE);
        }
    }

Looking forward to your reply as soon as possible, thank you very much

**Comment by michaelklishin** [comment]
It looks like a single key can be updated. Feel free to submit a PR, although `WorkPool` hasn't changed in a long time and really testing for possible side effects can take a while.

**Reply by wangqiandeniangniang** [reply]
thank you so much
